### PR TITLE
Issue 57: Reorganise repo directory structure

### LINKED ISSUE #57: Reorganize repository based on current plans
There are a few issues with the current repo structure:

1. Our NSSP model started as a "demo," hence the `nssp_demo` folder, but it is now the main offering of the repo.
2. We are collecting a variety of other "demo" scripts, currently stored in the `notebooks` folder.

I would like to reorganize the repo to make things a bit more clear. I think I will have a better idea of how to do that once we have run things in azure.

### PR DESCRIPTION
**Pull request by SamuelBrand1**
This PR closes #57.

Reorganises the repo directory structure as planned in #57:

- `demos` dir has the hosp-only model as a subdir.
- `pipelines` has the pipeline `nssp` as a subdir. IMO each pipeline should have a `tests` subdir eventually.
- `pyrenew-hew/utils.py` already existed.
- `scoringutilhelpers` has been removed as not used directly. These functions might end up in a `forecastoolsr` package, but at the moment are superceded by `R` scripts in the `nssp` pipeline.

NB: This PR has not moved any functions, I think we can sequence that over future issues.

### CODE REVIEW

**Comment by damonbayer** [comment]
Thanks @SamuelBrand1! Appreciate the chunking of these gargantuan tasks into manageable pieces.

I think  to the extent there are multiple pipelines, they should be for different tasks, e.g. real time forecasting vs retrospective evaluation, not different data streams.

**Comment by SamuelBrand1** [comment]
> Thanks @SamuelBrand1! Appreciate the chunking of these gargantuan tasks into manageable pieces.
> 
> I think to the extent there are multiple pipelines, they should be for different tasks, e.g. real time forecasting vs retrospective evaluation, not different data streams.

I think we might want _both_, or an update in the structure of the code. At the moment its set up to do multiple pathogens on the same data stream(s), but I think the easiest way to split is a seperate pipeline for

- `nssp`
- `nssp` + `ww`

With space to add hosp stream.

The RT forecasting should be an evolving structure? We've worked hard for the retro to use vintaged data, which means that is logically flows to have the "now" vintage.

File: demos/hosp_only_ww_model/hosp_only_ww_model.md
Comment (damonbayer) [inline demos/hosp_only_ww_model/hosp_only_ww_model.md:1]: delete this?

**Review by damonbayer — APPROVED** [review]
Thanks @SamuelBrand1!

**Pull request merged**